src/test/java/io/airlift/slice/TestSliceCompactFlag.java

```diff
@@ -0,0 +1,155 @@
+    {
+        assertCompact(new Slice());
+
+        boolean[] booleanArray = {true, false, false, true, false, true};
+        assertCompact(new Slice(booleanArray, 0, booleanArray.length));
+        assertNotCompact(new Slice(booleanArray, 0, booleanArray.length - 1));
+        assertNotCompact(new Slice(booleanArray, 1, booleanArray.length - 1));
+
+        byte[] byteArray = {(byte) 0, (byte) 1, (byte) 2, (byte) 3, (byte) 4, (byte) 5};
+        assertCompact(new Slice(byteArray));
+        assertCompact(new Slice(byteArray, 0, byteArray.length));
+        assertNotCompact(new Slice(byteArray, 0, byteArray.length - 1));
+        assertNotCompact(new Slice(byteArray, 1, byteArray.length - 1));
+
+        short[] shortArray = {(short) 0, (short) 1, (short) 2, (short) 3, (short) 4, (short) 5};
+        assertCompact(new Slice(shortArray, 0, shortArray.length));
+        assertNotCompact(new Slice(shortArray, 0, shortArray.length - 1));
+        assertNotCompact(new Slice(shortArray, 1, shortArray.length - 1));
+
+        int[] intArray = {0, 1, 2, 3, 4, 5};
+        assertCompact(new Slice(intArray, 0, intArray.length));
+        assertNotCompact(new Slice(intArray, 0, intArray.length - 1));
+        assertNotCompact(new Slice(intArray, 1, intArray.length - 1));
+
+        long[] longArray = {0L, 1L, 2L, 3L, 4L, 5L};
+        assertCompact(new Slice(longArray, 0, longArray.length));
+        assertNotCompact(new Slice(longArray, 0, longArray.length - 1));
+        assertNotCompact(new Slice(longArray, 1, longArray.length - 1));
+
+        float[] floatArray = {0f, 1f, 2f, 3f, 4f, 5f};
+        assertCompact(new Slice(floatArray, 0, floatArray.length));
+        assertNotCompact(new Slice(floatArray, 0, floatArray.length - 1));
+        assertNotCompact(new Slice(floatArray, 1, floatArray.length - 1));
+
+        double[] doubleArray = {0.0, 1.0, 2.0, 3.0, 4.0, 5.0};
+        assertCompact(new Slice(doubleArray, 0, doubleArray.length));
+        assertNotCompact(new Slice(doubleArray, 0, doubleArray.length - 1));
+        assertNotCompact(new Slice(doubleArray, 1, doubleArray.length - 1));
+    }
+
+    @Test
+    public void testSubSliceAndCopy()
+    {
+        byte[] byteArray = {(byte) 0, (byte) 1, (byte) 2, (byte) 3, (byte) 4, (byte) 5};
+        Slice slice = new Slice(byteArray);
+        assertCompact(slice.slice(0, slice.length()));
+        assertCompact(Slices.copyOf(slice));
+        assertCompact(Slices.copyOf(slice, 0, slice.length() - 1));
+        assertCompact(Slices.copyOf(slice, 1, slice.length() - 1));
+
+        Slice subSlice1 = slice.slice(0, slice.length() - 1);
+        Slice subSlice2 = slice.slice(1, slice.length() - 1);
+        assertNotCompact(subSlice1);
+        assertNotCompact(subSlice2);
+        assertCompact(Slices.copyOf(subSlice1));
+        assertCompact(Slices.copyOf(subSlice2));
+    }
+
+    @Test
+    public void testWrapDirectBuffer()
+            throws Exception
+    {
+        /**
```

Comment:
Let's make this `//` since it's a normal comment and not Javadoc. We could use `/*` instead of `/**` but our convention is to use `//`.